waku/v2/node/wakunode2.nim

```diff
@@ -648,195 +639,252 @@ proc connectToNodes*(n: WakuNode, nodes: seq[PeerInfo]) {.async.} =
-    else:
-      sqliteDatabase = dbRes.value
-
-  if not sqliteDatabase.isNil:
-    var  migrationPath = ""
-    if conf.persistPeers and conf.persistMessages: migrationPath = migration_types.ALL_STORE_MIGRATION_PATH
-    elif conf.persistPeers: migrationPath = migration_types.PEER_STORE_MIGRATION_PATH
-    elif conf.persistMessages: migrationPath = migration_types.MESSAGE_STORE_MIGRATION_PATH
-
-    # run migration 
-    info "running migration ... "
-    let migrationResult = sqliteDatabase.migrate(migrationPath)
-    if migrationResult.isErr:
-      warn "migration failed"
-    else:
-      info "migration is done"
+      if conf.storenode != "":
+        setStorePeer(node, conf.storenode)
 
-  var pStorage: WakuPeerStorage
+    # NOTE Must be mounted after relay
+    if (conf.lightpushnode != "") or (conf.lightpush):
+      mountLightPush(node)
 
-  if conf.persistPeers and not sqliteDatabase.isNil:
-    let res = WakuPeerStorage.new(sqliteDatabase)
-    if res.isErr:
-      warn "failed to init new WakuPeerStorage", err = res.error
-      waku_node_errors.inc(labelValues = ["init_store_failure"])
-    else:
-      pStorage = res.value
+      if conf.lightpushnode != "":
+        setLightPushPeer(node, conf.lightpushnode)
+    
+    # Filter setup. NOTE Must be mounted after relay
+    if (conf.filternode != "") or (conf.filter):
+      mountFilter(node)
 
-  let
-    (extIp, extTcpPort, extUdpPort) = setupNat(conf.nat, clientId,
-      Port(uint16(conf.tcpPort) + conf.portsShift),
-      Port(uint16(conf.udpPort) + conf.portsShift))
-    ## @TODO: the NAT setup assumes a manual port mapping configuration if extIp config is set. This probably
-    ## implies adding manual config item for extPort as well. The following heuristic assumes that, in absence of manual
-    ## config, the external port is the same as the bind port.
-    extPort = if extIp.isSome() and extTcpPort.isNone(): some(Port(uint16(conf.tcpPort) + conf.portsShift))
-              else: extTcpPort
-    node = WakuNode.new(conf.nodekey,
-                        conf.listenAddress, Port(uint16(conf.tcpPort) + conf.portsShift), 
-                        extIp, extPort,
-                        pStorage)
-
-  waitFor node.start()
-
-  if conf.swap:
-    mountSwap(node)
-
-  # TODO Set swap peer, for now should be same as store peer
-
-  # Store setup
-  if (conf.storenode != "") or (conf.store):
-    var store: WakuMessageStore
-    if (not sqliteDatabase.isNil) and conf.persistMessages:
-
-      let res = WakuMessageStore.init(sqliteDatabase)
-      if res.isErr:
-        warn "failed to init WakuMessageStore", err = res.error
-        waku_node_errors.inc(labelValues = ["init_store_failure"])
-      else:
-        store = res.value
+      if conf.filternode != "":
+        setFilterPeer(node, conf.filternode)
 
-    mountStore(node, store, conf.persistMessages)
+  # 4/6 Start node and mounted protocols
+  proc startNode*(node: WakuNode, conf: WakuNodeConf) =
```

Comment:
Since these (and other functions) are external functions, might want to update Nim Node API https://github.com/status-im/nim-waku/blob/master/docs/api/v2/node.md

Also, CHANGELOG!

Reply:
> Since these (and other functions) are external functions, might want to update Nim Node API

Ah, they shouldn't be public. Thanks! Had them in external module first (which caused a 100 issues) and forgot to remove the `*` after moving them back. Changed in 4cca119

> Also, CHANGELOG!

Done in 4cca119